plugins/locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/mylocationlayer/MyLocationLayerPlugin.java

```diff
@@ -0,0 +1,525 @@
+    compassListener = new CompassManager(mapView.getContext(), this);
+  }
+
+  /**
+   * Get the current {@link MyLocationLayerOptions} object which can be used to customize the look of the location icon.
+   *
+   * @return the {@link MyLocationLayerOptions} object this My Location layer plugin instance's using
+   * @since 0.1.0
+   */
+  public MyLocationLayerOptions getMyLocationLayerOptions() {
+    return options;
+  }
+
+  /**
+   * After creating an instance of this plugin, you can use this API to enable the location mode of your choice. These
+   * modes can be found in the {@link MyLocationLayerMode} class and the parameter only accepts one of those modes. Note
+   * that before enabling the My Location layer, you will need to ensure that you have the requested the required user
+   * location permissions.
+   * <p>
+   * <ul>
+   * <li>{@link MyLocationLayerMode#TRACKING}: Display the user location on the map as a small dot</li>
+   * <li>{@link MyLocationLayerMode#COMPASS}: Display the user location and current heading/bearing</li>
+   * <li>{@link MyLocationLayerMode#NAVIGATION}: Display the user location on the map using a navigation icon</li>
+   * <li>{@link MyLocationLayerMode#NONE}: Disable user location showing on the map</li>
+   * </ul>
+   *
+   * @param myLocationLayerMode one of the modes found in {@link MyLocationLayerMode}
+   * @since 0.1.0
+   */
+  public void setMyLocationEnabled(@MyLocationLayerMode.Mode int myLocationLayerMode) {
+    if (locationSource == null) {
+      locationSource = LocationSource.getLocationEngine(mapView.getContext());
+    }
+
+    if (myLocationLayerMode != MyLocationLayerMode.NONE) {
+      enableLocationUpdates();
+
+      options.initialize();
+
+      // Set an initial location if one available
+      Location lastLocation = locationSource.getLastLocation();
+
+      if (lastLocation != null) {
+        setLocation(lastLocation);
+        setAccuracy(lastLocation);
+      }
+
+      if (myLocationLayerMode == MyLocationLayerMode.COMPASS) {
+        setNavigationEnabled(false);
+        setMyBearingEnabled(true);
+      } else if (myLocationLayerMode == MyLocationLayerMode.NAVIGATION) {
+        setMyBearingEnabled(false);
+        setNavigationEnabled(true);
+      } else if (myLocationLayerMode == MyLocationLayerMode.TRACKING) {
+        setMyBearingEnabled(false);
+        setNavigationEnabled(false);
+      }
+    } else {
+      // Check that the mode isn't already none
+      if (this.myLocationLayerMode != MyLocationLayerMode.NONE) {
+        disableLocationUpdates();
+
+        options.removeLayerAndSources();
+      }
+    }
+    this.myLocationLayerMode = myLocationLayerMode;
+  }
+
+  /**
+   * Use to either force a location update or to manually control when the user location gets updated.
+   *
+   * @param location where you'd like the location icon to be placed on the map
+   * @since 0.1.0
+   */
+  public void setMyLocation(Location location) {
```

Comment:
Is null allowed here? Either annotate public api with Nullable or NonNull to make the purpose clear for the end user